go/pkg/tool/tool.go

```diff
@@ -245,42 +248,86 @@ func (c *Client) ShowAction(ctx context.Context, actionDigest string) (string, e
 		return "", err
 	}
 
+	if actionProto.Timeout != nil {
+		timeout, err := ptypes.Duration(actionProto.Timeout)
+		if err != nil {
+			return "", err
+		}
+		showActionRes.WriteString(fmt.Sprintf("Timeout: %s\n", timeout.String()))
+	}
+
 	commandProto := &repb.Command{}
 	cmdDg, err := digest.NewFromProto(actionProto.GetCommandDigest())
 	if err != nil {
 		return "", err
 	}
-	showActionRes.WriteString("Command\n========\n")
+	showActionRes.WriteString("Command\n=======\n")
 	showActionRes.WriteString(fmt.Sprintf("Command Digest: %v\n", cmdDg))
 
 	log.Infof("Reading command from action digest..")
 	if err := c.GrpcClient.ReadProto(ctx, cmdDg, commandProto); err != nil {
 		return "", err
 	}
+	for _, ev := range commandProto.GetEnvironmentVariables() {
+		showActionRes.WriteString(fmt.Sprintf("\t%s=%s\n", ev.Name, ev.Value))
+	}
 	cmdStr := strings.Join(commandProto.GetArguments(), " ")
 	showActionRes.WriteString(fmt.Sprintf("\t%v\n", cmdStr))
 
+	showActionRes.WriteString("\nPlatform\n========\n")
+	for _, property := range commandProto.GetPlatform().GetProperties() {
+		showActionRes.WriteString(fmt.Sprintf("\t%s=%s\n", property.Name, property.Value))
+	}
+
+	showActionRes.WriteString("\nInputs\n======\n")
 	log.Infof("Fetching input tree from input root digest..")
 	inpTree, _, err := c.getInputTree(ctx, actionProto.GetInputRootDigest())
 	if err != nil {
-		return "", err
+		showActionRes.WriteString("Failed to fetch input tree:\n")
+		showActionRes.WriteString(err.Error())
+		showActionRes.WriteString("\n")
+	} else {
+		showActionRes.WriteString(inpTree)
 	}
-	showActionRes.WriteString("\nInputs\n======\n")
-	showActionRes.WriteString(inpTree)
 
-	log.Infof("Fetching output tree from action result..")
-	outs, err := c.getOutputs(ctx, resPb)
-	if err != nil {
-		return "", err
+	if resPb == nil {
+		showActionRes.WriteString("\nNo action result in cache.\n")
+	} else {
+		log.Infof("Fetching output tree from action result..")
+		outs, err := c.getOutputs(ctx, resPb)
+		if err != nil {
+			return "", err
+		}
+		showActionRes.WriteString("\n")
+		showActionRes.WriteString(outs)
 	}
-	showActionRes.WriteString("\n")
-	showActionRes.WriteString(outs)
 	return showActionRes.String(), nil
 }
 
 func (c *Client) getOutputs(ctx context.Context, actionRes *repb.ActionResult) (string, error) {
 	var res bytes.Buffer
-	res.WriteString("Output Files:\n=============\n")
+
+	res.WriteString("------------------------------------------------------------------------\n")
+	res.WriteString("Action Result\n\n")
+	res.WriteString(fmt.Sprintf("Exit code: %d\n", actionRes.ExitCode))
```

Comment:
Note: tt's always zero, because failed actions are not cached. But it won't hurt displaying it.

Reply:
Hah, and I even just fixed the other bit of the code to not call this function in that scenario (because the ActionResult doesn't exist). Why do we even have that field?

Reply:
Because ActionResult doesn't always come from the remote cache, it is also returned by remote execution, where it can fail.